documentation/bolt_defaults_reference.md.erb

```diff
@@ -0,0 +1,92 @@
+# `bolt-defaults.yaml` options
+
+This page lists the configuration options that are available in the `bolt-defaults.yaml` configuration file,
+which is supported in both the [system-wide and user-level configuration directories](configuring_bolt.md).
+The avaiable options allow you to configure how Bolt runs and set the default configuration for each of the
+available transports.
+
+Options that are set in a `bolt-defaults.yaml` file in the system-wide configuration directory will apply
+to all Bolt projects run on the system. Likewise, options set in a `bolt-defaults.yaml` file in the user-level
+configuration directory apply to all Bolt projects run by the user.
+
+## Top-level options
+
+These are the configuration options that can be set at the top-level of the file.
+
+<% @top_level[:options].each do |option, desc| -%>
+### `<%= option %>`
+
+<%= desc %>
+
+<% if @top_level[:defaults][option] -%>
+- **Default:** <%= @top_level[:defaults][option] %>
+<% end %>
+<% end %>
+
+## Additional options
```

Comment:
Not thrilled with this header... 'Nested options', maybe?

Reply:
I haven't looked deeply enough into this, but I feel like this kind of indicates that the distinction isn't necessarily meaningful to users? I think we can either distinguish transport configuration from other configuration (so this would be 'Transport configuration', or get more granular like the help text and have a heading for 'groups' of config.

Reply:
Ya, the separate sections for sub-options felt a little strange. The current iteration lists the available sub-options under each option.